### ISSUE TITLE
java.lang.NullPointerException: null cannot be cast to non-null type java.time.Duration

### ISSUE THREAD

**Issue by siddjain** [jasync-sql/jasync-sql#306]
I upgraded to 2.1.0 from 2.0.8 and get this exception now at startup. 

```
Caused by: java.lang.NullPointerException: null cannot be cast to non-null type java.time.Duration
	at com.github.jasync.r2dbc.mysql.MysqlConnectionFactoryProvider.create(MysqlConnectionFactoryProvider.kt:77)
	at com.github.jasync.r2dbc.mysql.MysqlConnectionFactoryProvider.create(MysqlConnectionFactoryProvider.kt:23)
	at io.r2dbc.spi.ConnectionFactories.find(ConnectionFactories.java:110)
	at io.r2dbc.spi.ConnectionFactories.get(ConnectionFactories.java:140)
	at org.springframework.boot.r2dbc.ConnectionFactoryBuilder$OptionsCapableWrapper.buildAndWrap(ConnectionFactoryBuilder.java:191)
	at org.springframework.boot.r2dbc.ConnectionFactoryBuilder$PoolingAwareOptionsCapableWrapper.buildAndWrap(ConnectionFactoryBuilder.java:204)
	at org.springframework.boot.r2dbc.ConnectionFactoryBuilder.build(ConnectionFactoryBuilder.java:177)
	at org.springframework.boot.autoconfigure.r2dbc.ConnectionFactoryConfigurations.createConnectionFactory(ConnectionFactoryConfigurations.java:68)
	at org.springframework.boot.autoconfigure.r2dbc.ConnectionFactoryConfigurations$PoolConfiguration$PooledConnectionFactoryConfiguration.connectionFactory(ConnectionFactoryConfigurations.java:92)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:154)
	... 86 common frames omitted
```

my config:

```
spring.r2dbc.url=r2dbc:mysql://127.0.0.1:13306/xxx
spring.r2dbc.username=xxx
spring.r2dbc.password=xxx
spring.r2dbc.pool.initial-size=100
spring.r2dbc.pool.max-size=500
spring.r2dbc.pool.max-idle-time=30m
spring.r2dbc.pool.validation-query=SELECT 1
```

it works with 2.0.8

**Comment by oshai** [comment]
It looks like this is because query timeout config was considered non-nullable.
Fix in #307.

As a workaround you can try to add query timeout to the config.

**Comment by siddjain** [comment]
Thanks how do I add query timeout from the Java properties configuration file? Been trying that and not able to get it to work

Sent from my iPhone

On Sep 18, 2022, at 10:13 PM, Ohad Shai ***@***.***> wrote:

﻿

Closed #306<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fjasync-sql%2Fjasync-sql%2Fissues%2F306&data=05%7C01%7C%7C073dbff28689435048b608da99fdb51d%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637991612185550500%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=KnnLmbX6rYuHDgOZlARrkwBWkNTSVWha6sFa2xn2bH8%3D&reserved=0> as completed via 4920c78<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fjasync-sql%2Fjasync-sql%2Fcommit%2F4920c780c71321450ceabc1145d4b22447344abc&data=05%7C01%7C%7C073dbff28689435048b608da99fdb51d%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637991612185550500%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=6hZClEnv5TOA%2FPAj7QF4dtfVKIsxCTWJQ3DJYA4r9VI%3D&reserved=0>.
